sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -36,17 +36,25 @@ import org.apache.spark.sql.types.DataType
  *                    better to use Option of Seq[DataType] so we can use "None" as the case for no
  *                    type coercion. However, that would require more refactoring of the codebase.
  * @param udfName   The user-specified name of this UDF.
+ * @param udfDeterministic  True if the UDF is deterministic. Deterministic UDF returns same result
+ *                          each time it is invoked with a particular input.
+ * @param distinctLike  A UDF is considered distinctLike if the UDF can be evaluated on just the
+ *                      distinct values of a column.
```

Comment:
Hive UDFs ignores `distinctLike` and AFAIK there is no optimisation rules for `distinctLike` though, do we need this `param` now?